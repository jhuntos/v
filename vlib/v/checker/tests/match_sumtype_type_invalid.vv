type IoS = int | string

interface Animal {
	speak()
}

struct Dog {}

fn (d Dog) speak() {}

struct Cat {}

fn main() {
	match IoS(1) {
		byte {
			println('not cool')
		}
	}
	a := Animal(Dog{})
	match a {
		Cat {
			println('not cool either')
		}
		else {}
	}
}
