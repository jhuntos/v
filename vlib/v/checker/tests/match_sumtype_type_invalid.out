vlib/v/checker/tests/match_sumtype_type_invalid.vv:14:2: error: cannot use type `IoS` as type `byte`
   12 |
   13 | fn main() {
   14 |     match IoS(1) {
      |     ~~~~~~~~~~~~~~
   15 |         byte {
   16 |             println('not cool')
vlib/v/checker/tests/match_sumtype_type_invalid.vv:21:3: error: `Cat` doesn't implement method `speak`
   19 |     a := Animal(Dog{})
   20 |     match a {
   21 |         Cat {
      |         ~~~~~
   22 |             println('not cool either')
   23 |         }
